### ISSUE TITLE
New Release incoming?

### ISSUE THREAD

**Issue by ms178** [Alex313031/Thorium#100]
First of all, I love Thorium - your build is outperforming others by miles! Since you got Google Sync working, I am using Thorium as my main browser. I appreciate your time and effort that you put into this project! 

I saw new committs three weeks ago, but no new release emerged from that. As there were some security issues uncovered in Chrome in general since the last realease and as there are still some known bugs in both the AVX and AVX2 version of Thorium specifically, I am a bit worried about the current state of two and a half months of silence.

I'd also be more patient if there were some news entries in your dev blog explaining the silence. But I haven't seen any communication there either. I hope you'll find the time and motivation to maintain this awesome project.

**Comment by Alex313031** [comment]
@ms178 Working on special windows 7/8 builds, since chromium is ending support for those. I haven't deserted the project. I know I need to make some fresh releases soon.

**Comment by ms178** [comment]
Thanks a lot for the news! However, I wonder why support for these two EOL'ed operating systems is a priority over the newer releases though. People won't have too much fun with all the accumulated security issues that won't get fixed regardless of how secure the browser is that's being used, dropping support is the logical next step in my eyes.

**Comment by brucethemoose** [comment]
Well justified or not there are still a good number of W7 users out there.

**Comment by ms178** [comment]
@brucethemoose This is a question about the priotization of developer ressources and putting any effort into EOL'ed platforms is a unnessecary distraction in my eyes for the reasons already given. More so when the two more important main builds are already two and a half months without any new release and in need for timely updates.

**Comment by insilications** [comment]
I think the the question is relevant (specially the bit about accumulating security updates) . But ultimately, this is Alex's decision and we will respect that.

**Comment by ovidijusr** [comment]
Just wanted to say that I really love what you did with thorium and can't wait for a new version for mac arm cpus <3

**Comment by ms178** [comment]
As our calls have been heared and a new release came out, I am closing this issue.

**Comment by Alex313031** [comment]
@ovidijusr @brucethemoose @ms178 @insilications I still use XP, and 7, on top of Win 10, 11, and various linux distros. Many people still use 7 for the lower latency, less bloat, the beautiful Aero design, old hardware, directx 9 games, and others. I have a patch for ESU updates that I will put on my site. They just stopped esu updates for Win7 and 8 on Jan. 10 2023. So it will not be as outdated as some people thing it is. There are many arguments to be had for staying on an old platform, as well as for upgrading. I simply want to provide people the choice, and provide support. Next release will be based on M110. Also @ovidijusr @midzer will be making x64 and arm64 Mac releases soon.

**Comment by ms178** [comment]
@Alex313031 Your project, your choice. But to be honest, I'd rather like to see a x86-64-v3-Linux build.

**Comment by Alex313031** [comment]
@ms178 You mean an avx2 build?

**Comment by ms178** [comment]
@Alex313031 x86-64-v3 enables more than just AVX2 (AVX, AVX2, BMI1, BMI2, F16C, FMA, LZCNT, MOVBE, XSAVE + older ISAs) and aims to be a more optimized baseline for a more modern system with compatibility to both Intel and AMD (Intel Haswell+ / AMD Excavator+). You simply set `-march=x86-64-v3` as the target, which is supported by recent GCC and Clang compilers.

If you want to go even further, Haswell and Excavator share some more instructions that could be used as a superset. The overall approach gained some traction on Linux, with every major distribution considering such a move (albeit some aim for broader compatibility and want to stick to x86-64-v2 initially), e.g. CachyOS and ALHP already offer x86-64-v3 repos for Arch and offer a decent performance advantage over stock.

**Comment by Alex313031** [comment]
@ms178 Yeah thats what I meant. And thanks for the second comment. I had come across cachyos and was pleasantly surprised by the use of -O3, LTO, and -march flags just like thorium, but then I forgot the distro's name. So you reminded me lol. About ALHP, I didn't know about that, but I'm glad you shared that too. I might drop in the repos as a replacement in a VM, and if all goes well, then actually convert my ArchLabs drive to use these repos. I've made some pacman changes though, so I'll have to check that out.

**Comment by ptr1337** [comment]
> @ms178 Yeah thats what I meant. And thanks for the second comment. I had come across cachyos and was pleasantly surprised by the use of -O3, LTO, and -march flags just like thorium, but then I forgot the distro's name. So you reminded me lol. About ALHP, I didn't know about that, but I'm glad you shared that too. I might drop in the repos as a replacement in a VM, and if all goes well, then actually convert my ArchLabs drive to use these repos. I've made some pacman changes though, so I'll have to check that out.
> 
> I originally planned Thorium to only be AVX, it was only after lots of requests that I started making SSE3 builds, and AVX2 builds for windows. I could make a v3 build and post it in thorium-special for you to test.

Hi, CachyOS Maintainer here.

Since there are a bunch of request for a PKGBUILD for archlinux to build thorium, it would be a great idea to do this.
I considered to do a PKGBUILD for thorium based on the ungoogled-chromium PKGBUILD (https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ungoogled-chromium), but it is actually not very clear to me, which patches needs to be exactly applied, ...
Anyways - if there is once a working PKGBUILD we would cover the AVX2 builds for archlinux based systems. :)

ungoogled-chromium already gets build against -march=x86-64-v3, so it would be then thorium.

Regards

**Comment by ms178** [comment]
@Alex313031 As you see, there is quite some interest in both a binary build and source package. Thorium fits the spirit of CachyOS perfectly to offer more performance-optimized packages targeting modern hardware for the Linux community. Count me in for testing such a build. :)

@ptr1337 I'd suggest to prioritize to offer stable branch releases in such a CachyOS package first. Thorium was targeting the bleeding edge Chromium in the past and the latest package release was an exception to that; Alex announced to target the Beta and Dev branches in the future again. For the needs of CachyOS, I'd play it safe and avoid the risk of too many Choromium upstream bugs. If there is enough interest and infrastructure bandwith, there could be Beta and Dev builds later on.

**Comment by Alex313031** [comment]
@ptr1337 There have been requests? And there already is a PKGBUILD in the thorium repo, and there is the one used for thorium-bin in the AUR. I could guide on how to build it from source. But maybe I could just compile it remotely. Like do you guys have a build server where one could SSH in the just type the commands to build a -march=x86-64-v3 build.
And @ms178 the current thorium is based on stable. The next one will actually will be based on stable M110 too.

**Comment by ms178** [comment]
@Alex313031 Great, as the last release announcement pointed in the other direction. :) 

I guess ptr1337 meant something different. He wants to create a PKGBUILD for Arch which compiles Thorium from sources to provide essentially a march=x86-64-v3 and possibly march=x86-64-v4 builds for CachyOS directly on his own build server and making it easier for others to build Thorium locally (also using your advanced LLVM/Clang/Polly flags?), not a PKGBUILD that just downloads and installs the deb binary (that's what I currently provide in my repo and is also provided in the AUR already). I am sure you both can make it work.
